src/gpuhunt/providers/vultr.py

```diff
@@ -0,0 +1,205 @@
+                + other_plans_response.json().get("plans", [])
+            ),
+            "plans_metal": bare_metal_plans_response.json().get("plans_metal", []),
+        }
+        return convert_response_to_raw_catalog_items(combined_response)
+    except requests.RequestException as e:
+        logger.error(f"Failed to fetch plans: {str(e)}")
+        return None
+
+
+def convert_response_to_raw_catalog_items(response: dict) -> list[RawCatalogItem]:
+    catalog_items = []
+    plans = response.get("plans", []) + response.get("plans_metal", [])
+
+    for plan in plans:
+        for location in plan.get("locations", []):
+            if plan in response.get("plans_metal", []):
+                catalog_item = get_bare_metal_plans(plan, location)
+            else:
+                catalog_item = get_instance_plans(plan, location)
+            if catalog_item:
+                catalog_items.append(catalog_item)
+
+    return catalog_items
+
+
+def get_bare_metal_plans(plan: dict, location: str) -> Optional[RawCatalogItem]:
+    gpu_name, gpu_count = extract_gpu_info_from_id(plan.get("id", ""))
+    if gpu_name in EXCLUSION_LIST:
+        logger.info(f"Excluding plan with GPU {gpu_name} as it is not supported.")
+        return None
+    gpu_memory = get_gpu_memory(gpu_name) * gpu_count if gpu_name else None
+    gpu_vendor = get_gpu_vendor(gpu_name)
+    return RawCatalogItem(
+        instance_name=plan.get("id"),
+        location=location,
+        price=round(plan.get("monthly_cost", 0) / 730, 2),
+        cpu=plan.get("cpu_count"),
+        memory=plan.get("ram", 0) / 1024,
+        gpu_count=gpu_count,
+        gpu_name=gpu_name,
+        gpu_memory=gpu_memory,
+        gpu_vendor=gpu_vendor,
+        spot=False,
+        disk_size=plan.get("disk", 0),
+    )
+
+
+def get_instance_plans(plan: dict, location: str) -> Optional[RawCatalogItem]:
+    plan_type = plan.get("type", "")
+    if plan_type in ["vc2", "vhf", "vhp", "voc"]:
+        return RawCatalogItem(
+            instance_name=plan.get("id"),
+            location=location,
+            price=plan.get("hourly_cost", 0),
+            cpu=plan.get("vcpu_count"),
+            memory=plan.get("ram", 0) / 1024,
+            gpu_count=0,
+            gpu_name=None,
+            gpu_memory=None,
+            gpu_vendor=None,
+            spot=False,
+            disk_size=plan.get("disk", 0),
+        )
+    elif plan_type == "vcg":
+        gpu_name = (
+            plan.get("gpu_type", "").split("_")[1] if "_" in plan.get("gpu_type", "") else None
+        )
+        if gpu_name in EXCLUSION_LIST:
+            logger.info(f"Excluding plan with GPU {gpu_name} as it is not supported.")
+            return None
+        gpu_vendor = get_gpu_vendor(gpu_name)
+        gpu_memory_gb = plan.get("gpu_vram_gb", 0)
+        gpu_count = (
+            max(1, gpu_memory_gb // get_gpu_memory(gpu_name)) if gpu_name else 0
+        )  # For fractional GPU,
```

Comment:
Was this supposed to be `min(1, ...)`? Otherwise the only possible outcomes are 1 and 0

Reply:
Why max(1, ...) is correct:
1. If the division results less than 1 as in case of fractional GPUs, gpu_count = 1.
2. If the result is 1 or greater as in case of discrete GPUs, gpu_count = integer value after division.

Reply:
Yes, sorry, you are right